### PR TITLE
feat(performance) Add user_misery function to performance

### PR DESCRIPTION
**Pull request by evanh**
Instead of impact, calculate how many unique users were outside the apdex range,
as well as the count of total unique users, to give a sense of the impact a
poorly performing transaction is having on users.

### CODE REVIEW

File: src/sentry/api/event_search.py [hunk 1]
```diff
@@ -1106,6 +1107,13 @@ def has_default(self, params):
         "transform": u"plus(minus(1, divide(plus(countIf(less(duration, {satisfaction:g})),divide(countIf(and(greater(duration, {satisfaction:g}),less(duration, {tolerated:g}))),2)),count())),multiply(minus(1,divide(1,sqrt(uniq(user)))),3))",
         "result_type": "number",
     },
+    "user_misery": {
+        "name": "user_misery",
+        "args": [NumberRange("satisfaction", 0, None)],
+        "calculated_args": [{"name": "tolerated", "fn": lambda args: args["satisfaction"] * 4.0}],
+        "transform": u"uniqIf(user, duration > {tolerated:g})",
```
Comment (markstory) [inline src/sentry/api/event_search.py:1114]: Do we want fractional users if they had a somewhat bad response? That was in the design of impact() and was something David wanted to keep in the replacement. If we don't want to have 9.2 users frustrated we could round the values up and assume any partials count as fully frustrated.

Comment (evanh) [reply src/sentry/api/event_search.py:1114]: Like count users who had satisfied < duration < tolerated as .5, same as apdex?

Comment (markstory) [reply src/sentry/api/event_search.py:1114]: Yeah. Getting half points for users who had durations > acceptable & <= tolerated would be good. That would let slight misery to compound in aggregate.

Comment (evanh) [reply src/sentry/api/event_search.py:1114]: Per our offline discussion, I don't think there's a way to do this without losing the uniqueness of the users. So either we leave it as is, and have a unique count of users, or change it to include "annoyed" users but end up with a number that is more abstract.

**Review by markstory — APPROVED** [review]
Looks good to me. We can refine the rendering separately.

**Review by dashed — APPROVED** [review]

**Pull request merged**